### ISSUE TITLE
Stopping back-prop when using tf.cond

### ISSUE THREAD

**Issue by DavidBegert** [tensorflow/tensorflow#17962]
I am trying to stop back-prop from happening through one branch of a tf.cond statement. I am doing two forward passes through a CNN and trying to only back-prop the one with lower TCE (train_cross_entropy). The code is as follows: 
```    
def f1(): 
        with tf.control_dependencies([tf.stop_gradient(train_cross_entropy2)]):
            return train_cross_entropy1
def f2():
        with tf.control_dependencies([tf.stop_gradient(train_cross_entropy1)]):
            return train_cross_entropy2
train_cross_entropy = tf.cond(train_cross_entropy1 < train_cross_entropy2, f1, f2) 
```
The speed of this approach is equivalent to writing: 
```
train_cross_entropy = tf.add(train_cross_entropy1, train_cross_entropy2)
```
Whereas I would hope that the speed would be more similar to
```
train_cross_entropy = tf.add(train_cross_entropy1, tf.stop_gradient(train_cross_entropy2))
```
which happens to compute gradients about twice as fast.
I opened up a stack overflow question (https://stackoverflow.com/questions/49436264/proper-use-of-tf-cond-in-cnn/49439351) but I haven't found any help and I am wondering if this is the expected behaviour? 
How do I go about stopping gradients from being calculated backwards through both parts of the graph? 
Thanks!

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by jart** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!
